pandas/_libs/missing.pyx

```diff
@@ -44,7 +44,7 @@ cpdef bint checknull(object val):
     The difference between `checknull` and `checknull_old` is that `checknull`
     does *not* consider INF or NEGINF to be NA.
     """
-    return is_null_datetimelike(val, inat_is_null=False)
+    return val is C_NA or is_null_datetimelike(val, inat_is_null=False)
```

Comment:
I didn't move this yet into is_null_datetimelike (where all other checks happen), as right now the tslibs/nattype do not depend on missing.pyx or something from lib.pyx, and this might be on purpose?

Reply:
> right now the tslibs/nattype do not depend on missing.pyx or something from lib.pyx, and this might be on purpose?

Very much on purpose.  missing.pyx depends on nattype.pyx, and im saying it will need to depend on [wherever you're defining pd.NA] too.

Reply:
(now that ti look at this in the context of the diff its clear you already did what i had in mind)

Reply:
> Very much on purpose. missing.pyx depends on nattype.pyx, and im saying it will need to depend on [wherever you're defining pd.NA] too.

The long term problem, though, is to have `is_null_datetimelike` to recognize NA, it will need to depend on something (eg missing.pyx) outside of tslibs (as it would be strange to define NA in tslibs). 
Now, in practice, `is_null_datetimelike` is only used in period.pyx, so not a big priority to have NA working there.